qt/python/instrumentview/instrumentview/test/test_model.py

```
@@ -369,6 +369,21 @@ def test_picked_detector_ids(self):
+        model._is_masked = np.array([True, False, False])
+        model._detector_is_picked = np.array([True, False, True])
+        # The masked detector is not pickable, so it drops out of the mask entirely
+        np.testing.assert_array_equal(model.picked_detector_mask, [False, True])
+
+    def test_picked_detector_mask_round_trips_through_add_new_detector_key(self):
+        model, _ = self._setup_model([1, 2, 3])
+        model._cached_rois_map = {}
+        model._is_masked = np.array([True, False, False])
+        model._detector_is_picked = np.array([False, False, True])
+        model.add_new_detector_key(model.picked_detector_mask.tolist(), CurrentTab.Grouping)
+        np.testing.assert_array_equal(list(model._cached_rois_map.values())[0], np.array([False, False, True]))
```

Ruff (0.16.2)

[warning] Prefer `next(iter(model._cached_rois_map.values()))` over single element slice

Replace with `next(iter(model._cached_rois_map.values()))`

(RUF015)